crates/wasm-mutate/src/lib.rs

```diff
@@ -229,215 +235,142 @@ impl WasmMutate {
+                Payload::DataSection(reader) => {
+                    info.data = Some(info.raw_sections.len());
+                    info.section(SectionId::Data.into(), reader.range(), input_wasm);
+                }
+                Payload::CustomSection {
+                    name: _,
+                    data_offset: _,
+                    data: _,
+                    range,
+                } => {
+                    info.section(SectionId::Custom.into(), range, input_wasm);
+                }
+                Payload::AliasSection(reader) => {
+                    info.section(SectionId::Alias.into(), reader.range(), input_wasm);
                 }
+                Payload::UnknownSection {
+                    id,
+                    contents: _,
+                    range,
+                } => {
+                    info.section(id, range, input_wasm);
+                }
+                Payload::DataCountSection { count: _, range } => {
+                    info.section(SectionId::DataCount.into(), range, input_wasm);
+                }
+                Payload::Version { .. } => { }
+                Payload::End => { break; }
+                _ => todo!("{:?} not implemented", payload),
             }
-        });
-
-        if selected.len() == 0 { // return idem if no mutators
-            return Ok(input_wasm.to_vec())
+            wasm = &wasm[consumed..];
         }
 
+        Ok(info)
+    }
 
-        if cfg!(debug_assertions){
-            selected.iter().for_each(|(_, (range, name))| {
-                eprintln!("Applying {} on piece {:?}", name, range);
-            })
-        }
-
-        // Group selected mutations by range
-        // Grouping by range will allow to mutate over the same section after one mutation has already passed
-        type R = Box<dyn Fn(&WasmMutate, &[u8], &ModuleInfo) -> Vec<u8>>;
-        let mut ranges_to_mutate: HashMap<Range, Vec<&R>> = HashMap::new();
-        
-        selected.iter().for_each(|(muta, (range, name))|{
-            ranges_to_mutate.entry(*range).or_insert_with(|| Vec::new()).push(muta);
-        });
-
-        // Get the keys and sort them
-        let mut ranges = ranges_to_mutate.keys().collect::<Vec<&Range>>();
-
-        ranges.sort_by_key(|&&range| {
-            range.start
-        });
+    /// Run this configured `WasmMutate` on the given input Wasm.
+    pub fn run<'a>(&self, input_wasm: &'a [u8]) -> Result<Vec<u8>> {
+        let mut info = self.get_module_info(&input_wasm)?;
+
+        let mut mutators: Vec<Box<dyn Mutator>> = Vec::new();
+        initialize_and_filter! {
+            info,
+            self,
+            mutators,
+
+            RenameExportMutator{max_name_size: 100},
+            RemoveExportMutator,
+            ReturnI32SnipMutator,
+            SetFunction2Unreachable,
+        };
 
+        let mutator = mutators.choose(&mut self.get_rnd()).ok_or(Error::NoMutationsAplicable)?;
 
+        let module = mutator.mutate(&self, &mut info);
```

Comment:
I think the RNG should be pulled out into a local variable and then explicitly passed into the `mutate` method so that we are reusing the same RNG instead of calling `get_rnd()` again and recreating a new RNG from the same seed again (meaning that it will return the same first N random choices the last RNG instance did instead of continuing from where the last RNG instance left off).

Reply:
Addressed in PR 337